### ISSUE TITLE
TypeError: (0 , import_jsx_dev_runtime.jsxDEV) is not a function

### ISSUE THREAD

**Issue by binajmen** [remix-run/remix#5743]
### What version of Remix are you using?

1.14.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```diff
    "dev": "run-p dev:*",
+    "dev:build": "cross-env NODE_ENV=development remix dev",
+    "dev:serve": "cross-env NODE_ENV=development nodemon --watch build server.js",
-    "dev:server": "cross-env NODE_ENV=development node --inspect --require ./node_modules/dotenv/config --require ./mocks ./build/server.js",
-    "dev:build": "cross-env NODE_ENV=development npm run build:server -- --watch",
-    "dev:remix": "cross-env NODE_ENV=development remix watch",
    "dev:css": "cross-env NODE_ENV=development npm run generate:css -- --watch",
```

### Expected Behavior

HMR is working without errors

### Actual Behavior

Getting this error:
```
TypeError: (0 , import_jsx_dev_runtime.jsxDEV) is not a function
    at /Users//Projects//app/entry.server.tsx:64:9
    at new Promise (<anonymous>)
    at handleRequest (/Users//Projects//app/entry.server.tsx:59:10)
    at handleDocumentRequestRR (/Users//Projects//build/server.js:38925:18)
    at requestHandler (/Users//Projects//build/server.js:38787:22)
    at /Users//Projects//build/server.js:39945:26
```

**Comment by kiliman** [comment]
Yes, I'm getting that also with Remix App Server. Switching to Express works.

**Comment by MichaelDeBoey** [comment]
This will probably be fixed by @mcansh's #4358

**Comment by pcattori** [comment]
This was caused by `NODE_ENV` being set differently across different builds. The new dev server ensures `NODE_ENV=development` is set for _all_ builds while it is running. Fixed in v1.16.
